### PR TITLE
Jennyf/test md file

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: tools/ConfigureGeneratedApplications/Program.cs [hunk 1]
```diff
@@ -44,11 +44,8 @@ private static void GenerateIssueMdText(Configuration configuration, string fold
         {
             StringBuilder builder = new StringBuilder();
             builder.AppendLine("## Test the generated templates");
-            builder.AppendLine("1. Build the Microsoft.Identity.Web.sln solution");
-            builder.AppendLine(@"2. Add secrets to the `ProjectTemplates\configuration.json` file");
-            builder.AppendLine(@"3. Run the ProjectTemplates\test-templates.bat file");
-            builder.AppendLine(@"4. Open the `ProjectTemplates\bin\Debug\tests\tests.sln` solution");
-            builder.AppendLine("5. Test the following projects");
+            builder.AppendLine(@"1. Use `TESTING.md` file for steps for testing locally and for testing a release.");
```
Comment (jmprieur) [inline tools/ConfigureGeneratedApplications/Program.cs:47]: Maybe provide a link to TESTING.md ?

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# How to test the Microsoft.identity.Web project templates (locally or from a NuGet package)
+
+## Introduction
+### When to use this article?
+	- Before we release Microsoft.Identity.Web project templates (usually for each release of Microsoft.Identity.Web), we want to make sure that we test them from the release build.
```
Comment (jmprieur) [inline TESTING.md:5]: We don't want to indent here, as this transforms the text into code.

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+### When to use this article?
+	- Before we release Microsoft.Identity.Web project templates (usually for each release of Microsoft.Identity.Web), we want to make sure that we test them from the release build. 
+	- Before we commit changes in the project templates (under ProjectTemplates\templates) in the Microsoft.Identity.Web repo, we want to make sure that we test them in depth from the repo. Changes can be changes of Major versions of package reference of Microsoft.Identity.Web, sync from ASP.NET Core templates. This can also help detecting product bugs so we can do the same before releases of Microsoft.Identity.Web.
+	
+### When not to use this article?
+There is another article, explaining how to test templates from a NuGet package generated by ASP.NET Core team: How to test the ASP.NET core templates before they release. 
+
+## Principle:
+In this article you will:
+	- Configure the version of the templates to test by setting the ClientSemVer environment variable.
```
Comment (jmprieur) [inline TESTING.md:13]: Same thing: no indent

File: TESTING.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	- Manually test (for now) the generated projects.
+
+## How to generate the test projects?
+In a Developer Command Prompt:
+
+1. cd to the root of the repo (for instance `cd C:\gh\microsoft-identity-web`)
+
+2. Set the version of the templates to test.
+
+`Set ClientSemVer=0.4.0-preview`
```
Comment (jmprieur) [inline TESTING.md:27]: Indent by 3 spaces (so that it aligns with 2.)

**Review by jmprieur — COMMENTED** [review]
The text looks very good. thanks @jennyf19 
there are formating issues.

image

Do you mind if update your branch

File: tools/ConfigureGeneratedApplications/Program.cs (hunk 1 above)
Comment (jennyf19) [reply tools/ConfigureGeneratedApplications/Program.cs:47]: yeah, probably a good idea

**Pull request merged**